syntax = "proto3";

service TransferService {
  rpc MakeTransfer(TransferRequest) returns (TransferResponse);
  rpc GetHistory(UserRequest) returns (TransferHistoryResponse);
}

message TransferRequest {
  int32 from_account_id = 1;
  int32 to_account_id   = 2;
  double amount         = 3;
}

message TransferResponse {
  bool   success        = 1;
  string transaction_id = 2;
  string message        = 3;
}

message UserRequest {
  string user_id = 1;
}

message TransferItem {
  string timestamp      = 1;
  int32  from_account_id= 2;
  int32  to_account_id  = 3;
  double amount         = 4;
}

message TransferHistoryResponse {
  repeated TransferItem items = 1;
}
